tests/rptest/tests/upgrade_test.py

```diff
@@ -614,3 +614,37 @@ def cloud_log_size() -> int:
         # Assert manifest has not been mutated by comparing sizes
         total_cloud_size_after_upgrade = cloud_log_size()
         assert total_cloud_size_before_upgrade <= total_cloud_size_after_upgrade, f"Mismatch in cloud storage size after upgrade, bytes before: {total_cloud_size_before_upgrade} bytes_after: {total_cloud_size_after_upgrade}"
+
+
+class InstallSelfTest(RedpandaTest):
+    @cluster(num_nodes=1)
+    def test_redpandainstaller_install(self):
+        """
+        Smoke test that checks RedpandaInstaller.install
+        :return:
+        """
+        install = lambda version: self.redpanda._installer.install(
+            self.redpanda.nodes, version)
+        start = lambda: self.redpanda.start(self.redpanda.nodes[0:1])
+        get_version = lambda: self.redpanda.get_version(self.redpanda.nodes[0])
+        previous_version = lambda version: self.redpanda._installer.highest_from_prior_feature_version(
+            version)
+
+        head_version, head_version_str = install(RedpandaInstaller.HEAD)
+        self.logger.info(f"base step: install(HEAD) -> {head_version_str}")
+        start()
+        assert get_version() == head_version_str
+
+        # loop: run down the versions and check them
+        version = previous_version(head_version)
+        limit = 3
+        while limit > 0 and version:
+            line = version[0:2]
+            installed_version, installed_version_str = install(line)
+            self.logger.info(f"install {installed_version_str} from {line=}")
+            assert version == installed_version
+            start()
+            assert get_version() == installed_version_str
+            # eventually this will be None
```

Comment:
Curious about this: won't the versions be 22.3.x, 22.2.x, 22.1.x? Why will this return None?

Reply:
for example, given your list, previous_version((22,1)) -> None  (I don't remember what is the oldest release line available)

I don't expect for version to be None, since limit is only 3, but you never know

Reply:
I see. Thanks for removing this; even though the installer may return None in general, it shouldn't in this test so I found it confusing.